grid_map_pcl/src/grid_map_pcl_loader_node.cpp

```diff
@@ -15,34 +20,32 @@
 
 namespace gm = ::grid_map::grid_map_pcl;
 
-int main(int argc, char** argv) {
-  ros::init(argc, argv, "grid_map_pcl_loader_node");
-  ros::NodeHandle nh("~");
-  gm::setVerbosityLevelToDebugIfFlagSet(nh);
+int main(int argc, char ** argv)
+{
+  rclcpp::init(argc, argv);
+  auto node = std::make_shared<rclcpp::Node>("grid_map_pcl_loader_node");
+  gm::setVerbosityLevelToDebugIfFlagSet(node);
 
-  ros::Publisher gridMapPub;
-  gridMapPub = nh.advertise<grid_map_msgs::GridMap>("grid_map_from_raw_pointcloud", 1, true);
+  auto gridMapPub = node->create_publisher<grid_map_msgs::msg::GridMap>(
+    "grid_map_from_raw_pointcloud", rclcpp::QoS(1).transient_local());
 
-  grid_map::GridMapPclLoader gridMapPclLoader;
-  const std::string pathToCloud = gm::getPcdFilePath(nh);
+  grid_map::GridMapPclLoader gridMapPclLoader(node->get_logger());
+  const std::string pathToCloud = gm::getPcdFilePath(node);
   gridMapPclLoader.loadParameters(gm::getParameterPath());
   gridMapPclLoader.loadCloudFromPcdFile(pathToCloud);
 
-  gm::processPointcloud(&gridMapPclLoader, nh);
+  gm::processPointcloud(&gridMapPclLoader, node);
 
   grid_map::GridMap gridMap = gridMapPclLoader.getGridMap();
-  gridMap.setFrameId(gm::getMapFrame(nh));
-
-  gm::saveGridMap(gridMap, nh, gm::getMapRosbagTopic(nh));
-
-  //publish grid map
+  gridMap.setFrameId(gm::getMapFrame(node));
 
-  grid_map_msgs::GridMap msg;
-  grid_map::GridMapRosConverter::toMessage(gridMap, msg);
-  gridMapPub.publish(msg);
+  gm::saveGridMap(gridMap, node, gm::getMapRosbagTopic(node));
 
+  // publish grid map
+  auto msg = grid_map::GridMapRosConverter::toMessage(gridMap);
+  gridMapPub->publish(std::move(msg));
```

Comment:
Are you sure the `std::move` is doing the intended thing here? 

which `publish()` function would it use in this case? https://github.com/ros2/rclcpp/blob/master/rclcpp/include/rclcpp/publisher.hpp#L185

Reply:
The one you highlighted is the one being used.
Without `std::move` the code doesn't compile. What do you think might be wrong?

Reply:
All good, then.